docs/src/manuals/custom-node.md

```diff
@@ -0,0 +1,338 @@
+
+$$\vec{\nu}(x) \propto \exp \int q(\pi) \ln \mathrm{Ber}(x\mid \pi) \mathrm{d}x$$
+
+$$\overleftarrow{\nu}(\pi) \propto \exp \sum_{x \in \{0,1\}} q(x) \ln \mathrm{Ber}(x\mid \pi)$$
+
+These messages depend on the marginals on the adjacent edges and not on the incoming messages as was the case with sum-product message passing. Update rules that operate on the marginals instead of the incoming messages are specified with the `q_{interface}` argument names. With these update rules, we can often support a wider family of distributions. Below we directly give the variational update rules. Deriving them yourself will be a nice challenge.
+
+```julia
+#rules towards out
+@rule MyBernoulli(:out, Marginalisation) (q_π :: PointMass,) = MyBernoulli(mean(q_π))
+
+@rule Bernoulli(:out, Marginalisation) (q_π::Any,) = begin
+    rho_1 = mean(log, q_π)          # E[ln(x)]
+    rho_2 = mean(mirrorlog, q_π)    # E[log(1-x)]
+    m = max(rho_1, rho_2)
+    tmp = exp(rho_1 - m)
+    p = clamp(tmp / (tmp + exp(rho_2 - m)), tiny, one(m))
+    return Bernoulli(p)
+end
+
+#rules towards π
+@rule MyBernoulli(:π, Marginalisation) (q_out :: Any,) = begin
+    p = mean(q_out)
+    return Beta(one(p) + p, 2one(p) - p)
+end
+```
+
+
+
+In the example that we will show later on, we solely use sum-product message passing. Variational message passing requires us to set the local constraints in our model, something which is out of scope of this tutorial.
+
+
+---
+
+
+## Step 3: Defining joint marginals for our node
+
+
+The entire probabilistic model can be scored using the Bethe free energy, which bounds the log-evidence for acyclic graphs. This Bethe free energy consists out of the sum of node-local entropies, negative node-local average energies and edge specific entropies. Formally we can denote this by
+
+$$F[q,f] = - \sum_{a\in\mathcal{V}} \mathrm{H}[q_a(s_a)] - \sum_{a\in\mathcal{V}}\mathrm{E}_{q_a(s_a)}[\ln f_a(s_a)] + \sum_{i\in\mathcal{E}}\mathrm{H}[q_i(s_i)]$$
+
+Here we call $q_a(s_a)$ the joint marginals around a node and $-\mathrm{E}_{q_a(s_a)}[\ln f_a(s_a)]$ we term the average energy.
+
+In order to be able to compute the Bethe free energy, we need to first describe how to compute $q_a(s_a)$, defined in our case as 
+
+$$q(x_k, \pi) = \vec{\mu}(\pi) \overleftarrow{\mu}(x_k) \mathrm{Ber}(x_k \mid \pi)$$
+
+To calculate the updated posterior marginal for our custom distribution, we need to provide the posterior marginals for the interfaces of our node in a `NamedTuple`. In our case, the posterior marginal for the observation is still the same `PointMass` distribution. However, to calculate the posterior marginal over `π`, we use `RxInfer`'s built-in `prod` functionality to multiply the `Beta` prior with the `Beta` likelihood. This gives us the updated posterior distribution, which is also a `Beta` distribution. We use `ProdAnalytical()` parameter to ensure that we multiply the two distributions analytically. This is done as follows:
```

Comment:
It's actually a bad example, because we don't' need to return a `NamedTuple` in general. `NamedTuple` is returned if and only if the result is assumed to be factorized. E.g. if need to return `q(x, y)`, but we know that the result has the `q(x)q(y)` form we return a corresponding `NamedTuple`. But in general a user is supposed to compute `q(x, y)` as a single distribution. I think that should be clarified further. The reason here for the `NamedTuple` is because one of the arguments is a `PointMass` so we can safely factorize the marginal.

Reply:
My bad, it is explained better already in the next paragraph. But the wording that "we need to return a `NamedTuple`" is not correct and should be changed.